### PR TITLE
istio-default profile

### PR DESCRIPTION
**Pull request by gytis**

### CODE REVIEW

**Review by geoand — APPROVED** [review]

**Comment by metacosm** [comment]
@gytis are we able to override the enricher config? If yes, how?

File: src/main/resources/META-INF/fabric8/profiles.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    - f8-maven-issue-mgmt
+    # Dependencies shouldn't be enriched anymore, therefor it's last in the list
+    - fmp-dependency
+    - f8-watch
+    - fmp-revision-history
+    - fmp-docker-registry-secret
+    excludes:
+    - fmp-openshift-route
+    config:
+      fmp-istio-enricher:
```
Comment (metacosm) [inline src/main/resources/META-INF/fabric8/profiles.yml:45]: Looks good but I'm wondering if/how this config can be overridden in projects using the enricher. This needs to be checked before merging.

Comment (gytis) [reply src/main/resources/META-INF/fabric8/profiles.yml:45]: You can override them in plugin configuration:
```
         <plugin>
            <groupId>io.fabric8</groupId>
            <artifactId>fabric8-maven-plugin</artifactId>
            <configuration>
              <profile>istio-default</profile>
              <enricher>
                <config>
                  <fmp-istio-enricher>
                    <istioVersion>...</istioVersion>
                  </fmp-istio-enricher>
                </config>
              </enricher>
            </configuration>
          </plugin>
```

Comment (metacosm) [reply src/main/resources/META-INF/fabric8/profiles.yml:45]: 👍

Comment (metacosm) [reply src/main/resources/META-INF/fabric8/profiles.yml:45]: Is there a reason why the default istio version is 0.4.0? If not, I'd suggest using the latest version.

Comment (gytis) [reply src/main/resources/META-INF/fabric8/profiles.yml:45]: This I'm not exactly sure. 0.4.0 was in the previous examples of this enricher.

Which version are you currently using with the new boosters? I've tested mutual tsl on 0.4.0, but Libor is getting some issues with 0.6.0 on it.

Comment (cmoulliard) [reply src/main/resources/META-INF/fabric8/profiles.yml:45]: > 0.4.0

No but that was the version available when the enricher has been created.

If you target to use `0.6.0`, please verify like also for `0.5.0` that the args injected by ``istioctl inject ...` command are still the same. If this is not the case, then the class where such arguments are defined should be updated

Comment (gytis) [reply src/main/resources/META-INF/fabric8/profiles.yml:45]: So it seems that we should plan a separate task which is to upgrade the enricher to work with the newer versions.
I think we can keep 0.4.0 in the profiles.yml and update it once the whole enricher is verified to work with the newer version.

Comment (gytis) [reply src/main/resources/META-INF/fabric8/profiles.yml:45]: That would be a good opportunity to add some tests too.

Comment (metacosm) [reply src/main/resources/META-INF/fabric8/profiles.yml:45]: I will work on the enricher.

**Pull request merged**